### PR TITLE
PVS Enhancement - Phase 1

### PR DESCRIPTION
**Pull request by 0xABADCAFE**
**Main**
- Performs a per-shared-edge analysis for each Zone's PVS and restricts the set of zones that are considered to just those that can be reached via those edges. This solves a range of known overdraw cases and improves performance significantly.
- Provides a mechanism for precompiled "zone errata" that fixes issues with the PVS during level load.

Note that this is not a baseline performance increase, rather it deals with sudden frame rate dips caused by overdraw in scenes where areas of the map are drawn, then overdrawn by a closer wall.

**Sundry**
- Fixes fractional precision of FPS counter.
- Relocate FPS counter in release build.
- Added menu item to show FPS (release build, has no effect in dev mode as FPS is always shown), persisted in prefs.cfg
- Additional structure definition.

**TODO**
The following issues can be built on the work in this iteration:
- Expand the edge PVS to deal with lower and upper zone visibility.
- Introduce door/lift visibility.

### CODE REVIEW

**Comment by 0xABADCAFE** [comment]
Some of my commit message typos make it look like I've been abusing something other than the code!

File: ab3d2_source/c/system.c [hunk 1]
```diff
@@ -360,6 +362,11 @@ void Sys_FrameLap()
 
 void Sys_EvalFPS()
 {
+#ifndef DEV
```
Comment (mheyer32) [inline ab3d2_source/c/system.c:365]: I just _KNOW_ people will want to be able to see FPS, even in release mode.
TKG could become a benchmarking tool for people.

File: ab3d2_source/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Documentation
```
Comment (mheyer32) [inline ab3d2_source/docs/README.md:1]: Awesome! :-D

**Comment by mheyer32** [comment]
LGTM. I cannot repeat myself often enough: I love how the code gets more and more structured and better readable, yet we didn't loose perf (quite the opposite). This is more than I ever hoped for. Well done!

**Pull request merged**